### DISCUSSION TITLE
show rule with if condition

### DISCUSSION THREAD

**Discussion by 8LWXpg** [typst/typst#1416] [Q&A]
I store options in a dict called `option` and create show rules based on it.

Currently, I cannot pass show rules to the outer scope, so I use this workaround:

```
show: c => if option.ul {
	underline(c)
} else {
	c
}
show: c => if option.ol {
	overline(c)
} else {
	c
}
[#str]
```

 Is there a cleaner or better way to do this?

**Comment by Enivex** [comment]
I mean, it depends what you want to do, but one possible way is
```
#let effects = (underline,
                overline,
                text.with(fill: blue))
#let toggles = (true,
                false,
                true)
#show: effects
  .zip(toggles)
  .filter(t => t.last())
  .map(t => t.first())
  .fold( t => t, (a,f) => t => f(a(t)))

Hello world
```
![image](https://github.com/typst/typst/assets/2191754/cc4b2ff2-f46c-4666-8da9-d9efe26b571c)
